File: output/deepseek_chat/avatar/algo_based_translation/python/java/codeforces_108_B.java

import java.util.*;

public class codeforces_108_B {
    public static String s() {
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine();
        String[] inputArray = scanner.nextLine().split(" ");
        int[] a = new int[inputArray.length];
        for (int i = 0; i < inputArray.length; i++) {
            a[i] = Integer.parseInt(inputArray[i]);
        }
        Arrays.sort(a);
        for (int i = 1; i < a.length; i++) {
            if (a[i] != a[i - 1] && a[i] < a[i - 1] * 2) {
                return "YES";
            }
        }
        return "NO";
    }
    
    public static void main(String[] args) {
        System.out.println(s());
    }
}

File: output/deepseek_chat/avatar/algo_based_translation/java/python/codeforces_108_B.py

import sys

def main():
    data = sys.stdin.read().split()
    n = int(data[0])
    a = list(map(int, data[1:1+n]))
    a.sort()
    for i in range(len(a) - 1):
        if a[i+1] < a[i] * 2 and a[i] != a[i+1]:
            print("YES")
            return
    print("NO")

if __name__ == "__main__":
    main()

